specfile/specfile.py

```diff
@@ -120,18 +183,32 @@ def _dump_debug_info(self, message) -> None:
         )
 
     @staticmethod
-    def _read_lines(path: Path) -> Tuple[List[str], bool]:
-        content = path.read_text(encoding="utf8", errors="surrogateescape")
-        return content.splitlines(), content[-1] == "\n"
+    def _read_lines(stream: TextIO) -> Tuple[List[str], bool]:
+        stream.seek(0)
+        content = stream.read()
```

Comment:
The argument doesn't have to be `TextIO` (I'm surprised mypy didn't catch this):

```suggestion
    def _read_lines(file: IOBase) -> Tuple[List[str], bool]:
        file.seek(0)
        raw_content = file.read()
        if isinstance(raw_content, str):
            content = raw_content
        else:
            content = raw_content.decode("utf8", errors="surrogateescape")
```

Question is what to do with custom `TextIOBase`-based objects as we can't control their `encoding` and `errors` attributes. I suppose it's a responsibility of the user to make sure the spec file content can be properly decoded. So we should document that.